src/generic/GenericTypes.jl

```diff
@@ -421,21 +421,19 @@ end
    S::Vector{Symbol}
    ord::Symbol
    mpoly_ring::AbstractAlgebra.MPolyRing{T}
+   cached::Bool
 
    function UniversalPolyRing{T}(
-      R::Ring, internal_ordering::Symbol, cached::Bool=true
+      R::Ring, s::Vector{Symbol}, internal_ordering::Symbol, cached::Bool=true
    ) where {T<:RingElement}
       @assert elem_type(R) == T
-      return get_cached!(
-         UnivPolyID, (R, internal_ordering), cached
```

Comment:
If you don't use `UnivPolyID` anymore you should also delete is definition.

But instead why not keep it, but adjust its definition slightly, so that we can do here what @thofma suggested, i.e.:
```
      return get_cached!(
         UnivPolyID, (R, s, internal_ordering), cached
```